src/widget/wglwidgetqglwidget.cpp

```diff
@@ -0,0 +1,34 @@
+#include <QWindow>
+
+#include "waveform/sharedglcontext.h"
+#include "widget/wglwidget.h"
+
+WGLWidget::WGLWidget(QWidget* parent)
+        : QGLWidget(parent, SharedGLContext::getWidget()) {
+    setAttribute(Qt::WA_NoSystemBackground);
+    setAttribute(Qt::WA_OpaquePaintEvent);
+    setAutoFillBackground(false);
+    setAutoBufferSwap(false);
+    // Not interested in repaint or update calls, as we draw from the vsync thread
+    setUpdatesEnabled(false);
+}
+
+bool WGLWidget::isContextValid() const {
+    return context()->isValid();
+}
+
+bool WGLWidget::isContextSharing() const {
+    return context()->isSharing();
+}
+
+bool WGLWidget::shouldRender() const {
+    return true;
+    return isValid() && isVisible() && windowHandle() && windowHandle()->isExposed();
```

Comment:
This code cannot be reached.

Reply:
hm, this one has got me puzzled a bit. the second line seems very reasonable but i must have added the return true at some point to try something out. or maybe it's a merge gone wrong. i will try removing the first line and see if i notice anything strange.